src/gluonnlp/embedding/embed_container.py

```diff
@@ -0,0 +1,121 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=consider-iterating-dictionary, too-many-lines
+"""Load token embedding"""
+
+__all__ = [
+    'StaticEmbedding', 'FastText'
+]
+
+import io
+import logging
+import os
+import warnings
+import fasttext
+import numpy as np
+import mxnet as mx
+from .embed_loader import load_embeddings
+from .evaluation import CosineSimilarity, HyperbolicCosineSimilarity
+from ..data import Vocab
+
+class StaticEmbedding:
+    def __init__(self, vocab=None, matrix=None, embedding_space="euclidean"):
+        self.embedding_space = embedding_space
+        self.unk_token = unk_token
+        self._vocab = vocab
+        self._data = matrix
+
+    @property
+    def unk_method(self):
+        return None
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def vocab(self):
+        return self._vocab
+
+    def load(self, vocab=None, pretrained_name_or_dir='glove.6B.50d', unknown='<unk>', unk_method=None):
+        method = self.unk_method if self.unk_method is not None else unk_method
+        if vocab is None:
+            self._data, self._vocab = load_embeddings(vocab, pretrained_name_or_dir,
+                                                      unknown, unk_method=method)
+        else:
+            self._vocab = vocab
+            self._data = load_embeddings(vocab, pretrained_name_or_dir,
+                                         unknown=unknown, unk_method=self.method)
+        return self.data, self.vocab
+
+    def similarity(self, words1, words2, eps=1e-10):
```

Comment:
Should we follow the squad and glue design in that the evaluation functionality is located in one of the scripts folders? For example, https://github.com/dmlc/gluon-nlp/tree/numpy/scripts/question_answering

We should also support the analogy task, though the current PR doesn't need to provide all evaluation methods.